metricflow-semantics/metricflow_semantics/model/semantics/semantic_model_lookup.py

```diff
@@ -70,6 +70,11 @@ def __init__(self, model: SemanticManifest, custom_granularities: Dict[str, Expa
         # Cache for agg. time dimension for measure.
         self._measure_reference_to_agg_time_dimension_specs: Dict[MeasureReference, Sequence[TimeDimensionSpec]] = {}
 
+    @property
```

Comment:
I couldn't pass a set to `StructuredLinkableSpecName.from_name()` if we want to cache that, since sets aren't hashable so not cachable. I decided to use caching instead of changing this to a set.